WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -1158,38 +1154,12 @@ public void onClick(View v) {
             case RequestCodes.LOGIN_EPILOGUE:
                 if (resultCode == RESULT_OK) {
                     setSite(data);
-                    if (getMySiteFragment() != null) {
-                        showQuickStartDialog();
-                    } else {
-                        passOnActivityResultToMySiteFragment(requestCode, resultCode, data);
-                    }
+                    passOnActivityResultToMySiteFragment(requestCode, resultCode, data);
                 }
                 break;
             case RequestCodes.SITE_PICKER:
-                if (getMySiteFragment() != null) {
-                    boolean isSameSiteSelected = data != null
-                                                 && data.getIntExtra(
-                            SitePickerActivity.KEY_SITE_LOCAL_ID,
-                            SelectedSiteRepository.UNAVAILABLE
-                    ) == mSelectedSiteRepository.getSelectedSiteLocalId();
-
-                    if (!isSameSiteSelected) {
-                        QuickStartUtils.cancelQuickStartReminder(this);
-                        AppPrefs.setQuickStartNoticeRequired(false);
-                        AppPrefs.setLastSkippedQuickStartTask(null);
-                        mPrivateAtomicCookie.clearCookie();
```

Comment:
@ashiagr - Don't we still need to set `mPrivateAtomicCookie.clearCookie()` in this instance of a site change? (Maybe it's somewhere that I haven't seen yet - since I am reviewing commit by commit)

Reply:
Nice catch @zwarm! I've brought back the entire block in 15c664b. Seems like an existing bug where the code was not included for `ImprovedMySiteFragment`.